group_project_v2/stage_components.py

```diff
@@ -123,13 +123,15 @@ def validate_field_data(self, validation, data):
 class StaticContentBaseXBlock(BaseStageComponentXBlock, XBlockWithPreviewMixin, NoStudioEditableSettingsMixin):
     TARGET_PROJECT_NAVIGATOR_VIEW = None
     TEXT_TEMPLATE = None
+    TEMPLATE_PATH = "templates/html/components/static_content.html"
 
     def student_view(self, context):
-        activity = self.stage.activity
-        if activity.project.navigator is None:
-            return Fragment()
-
-        target_block = activity.project.navigator.get_child_of_category(self.TARGET_PROJECT_NAVIGATOR_VIEW)
+        try:
+            activity = self.stage.activity
+            target_block = activity.project.navigator.get_child_of_category(self.TARGET_PROJECT_NAVIGATOR_VIEW)
+        except AttributeError:
+            activity = None
+            target_block = None
```

Comment:
@smarnach none are optional, but might still not exist - in such case it needs returning empty fragment rather than crashing